test/t8_geometry/t8_gtest_geometry_lagrange.cxx

```diff
@@ -0,0 +1,259 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2023 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_gtest_geometry_lagrange.cxx
+* Provide tests to check the geometry mappings via Lagrange interpolation.
+* The tests are parametrized on the element type and the polynomial degree.
+*/
+
+#include <array>
+#include <cstdlib>
+#include <sstream>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <test/t8_gtest_macros.hxx>
+#include <t8_eclass.h>
+#include <t8_vec.h>
+#include <t8_element_cxx.hxx>
+#include <t8_cmesh.h>
+#include <t8_cmesh_vtk_writer.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+#include <t8_forest/t8_forest.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx>
+
+#define MAX_POLYNOMIAL_DEGREE 2
```

Comment:
Since this is valid in general, we should move this to `t8_geometry_lagrange.hxx`

Reply:
But `t8_geometry_lagrange` is not concerned about the testing, so why should the `gtest` header go there? Concerning the other includes: you are right that if I already include e.g. `t8_forest/t8_forest.h` in `t8_geometry_lagrange.hxx`, then there is no need to explicitly import it in this file. However, I adhered to the [include-what-you-use](https://include-what-you-use.org/) philosophy when I created this contribution.

Reply:
This comment was specifically about `#define MAX_POLYNOMIAL_DEGREE 2`, but GitHub also shows the other lines. Since this macro describes what is currently implemented, I would move this to `t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx`